### PR TITLE
Wrap the calls of map::at() in a try-catch block

### PR DESCRIPTION
**Pull request by StefanPalashev**

### CODE REVIEW

**Comment by f-squirrel** [comment]
Well, now I see where the PR comes from.

I think that wrapping map::at calls with a try-except block is not a good practice:
If the flow expects the keys to be absent, then we need to look for a value, and in case of its absence, log/throw an exception with a meaningful message.

If the flow does not expect the keys to be absent, then leave the `out_of_range` exception for higher-level methods.

File: bftengine/include/bftengine/CryptoManager.hpp [hunk 1]
```diff
@@ -54,8 +54,18 @@ class CryptoManager : public IKeyExchanger, public IMultiSigKeyGenerator {
...
+    auto log = logger();
+    LOG_INFO(log, "Generating new multisig key pair");
+
+    try {
+      auto cryptoSystem = cryptoSystems_.at(0);
+      return cryptoSystem->cryptosys_->generateNewKeyPair();
+    }
+    catch(const std::out_of_range& e) {
+      LOG_FATAL(log, e.what());
+      LOG_FATAL(log, "cryptoSystems_.at(0) has failed!");
```
Comment (yuliasherman) [inline bftengine/include/bftengine/CryptoManager.hpp:66]: Actually, you can combine those two in one log: LOG_FATAL(log, "cryptoSystems_.at(0) has failed! " << e.what);

File: bftengine/src/bftengine/ClientsManager.hpp [hunk 2]
```diff
@@ -82,7 +82,7 @@ class ClientsManager : public ResPagesClient<ClientsManager, 0> {
   // General
   static uint32_t reservedPagesPerClient(const uint32_t& sizeOfReservedPage, const uint32_t& maxReplySize);
   int getIndexOfClient(const NodeIdType& id) const;
-
+
```
Comment (yuliasherman) [inline bftengine/src/bftengine/ClientsManager.hpp:85]: I guess you touched this file occasionally?

File: bftengine/src/bftengine/KeyStore.cpp [hunk 3]
```diff
@@ -47,8 +47,19 @@ std::optional<ClusterKeyStore::PublicKeys> ClusterKeyStore::loadReplicaKeyStoreF
 }
 
 void ClusterKeyStore::saveReplicaKeyStoreToReserevedPages(const uint16_t& repID) {
```
Comment (yuliasherman) [inline bftengine/src/bftengine/KeyStore.cpp:49]: The function name is misleading.

File: communication/src/TlsConnectionManager.cpp [hunk 4]
```diff
@@ -128,6 +128,13 @@ void ConnectionManager::handleSend(const NodeNum destination, std::shared_ptr<Ou
   if (it != connections_.end()) {
     it->second->send(std::move(msg));
     status_->total_messages_sent++;
+    if (config_.statusCallback && isReplica() && (status_->total_messages_sent % 1000 == 1)) {
```
Comment (yuliasherman) [inline communication/src/TlsConnectionManager.cpp:131]: How does this change relate to .at() issue?

File: kvbc/test/sparse_merkle/test_db.h [hunk 5]
```diff
@@ -34,10 +34,19 @@ class TestDB : public concord::kvbc::sparse_merkle::IDBReader {
       return BatchedInternalNode();
     }
     auto root_key = InternalNodeKey::root(latest_version_);
-    return internal_nodes_.at(root_key);
+    return get_internal(root_key);
   }
 
-  BatchedInternalNode get_internal(const InternalNodeKey& key) const override { return internal_nodes_.at(key); }
```
Comment (yuliasherman) [inline kvbc/test/sparse_merkle/test_db.h:40]: We don't need to do this change for tests

File: bftengine/src/bftengine/ClientsManager.hpp (hunk 2 above)
Comment (StefanPalashev) [reply bftengine/src/bftengine/ClientsManager.hpp:85]: I was about to make a new function that had to do the same thing as the one above. It seems that I've left a blank space unintentionally.

File: bftengine/include/bftengine/CryptoManager.hpp (hunk 1 above)
Comment (StefanPalashev) [reply bftengine/include/bftengine/CryptoManager.hpp:66]: Thanks for pointing that out, I'll make the necessary changes.

**Review by teoparvanov — APPROVED** [review]
LTGM, thanks @StefanPalashev !

**Pull request merged**